pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -164,7 +164,15 @@ def optimize_budget_distribution(
 
     if budget_ranges is None:
         budget_ranges = {
-            channel: (0, min(total_budget, parameters[channel][0]))
+            channel: (
+                0,
+                min(
+                    total_budget,
+                    9 * parameters[channel][1]
```

Comment:
where does the `9` come from?

Reply:
Hey Will, the initial idea was to use the maximum budget or the point where saturation reaches 90% as the default optimization limit. However, I was passing this saturation value to 90% in Y, not X.

Why 9? If you consider Michaelis mente functions is `Y = alpha*X/lam+X` then if you want to find Y at 90% (0.9*Y) after simplifying the function you get X = 9 * lam. I left a small code snippet you can use to reproduce the results, and test if you want.

``` python
# Define parameters for Vmax and Km
alpha = 10
lam = 5

# Generate substrate concentrations (S) from 0 to 100
X = np.linspace(0, 100, 500)
y = michaelis_menten(X, alpha, lam)

# Calculate the substrate concentration at which the reaction rate is 90% of Vmax (Y = 9)
X_90_percent_example = 9 * lam

# Calculate the reaction rate at this substrate concentration
Y_90_percent_example = michaelis_menten(X_90_percent_example, alpha, lam)

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(X, y, label='Michaelis-Menten Curve', linewidth=2)
plt.scatter([X_90_percent_example], [Y_90_percent_example], color='red', zorder=5, label=f'90% of alpha (X={X_90_percent_example}, y={Y_90_percent_example})')
plt.axhline(y=alpha, color='g', linestyle='--', label=f'alpha = {alpha}')
plt.axhline(y=Y_90_percent_example, color='r', linestyle='--', label=f'90% of alpha = {Y_90_percent_example}')
plt.xlabel('Cost Units (X)')
plt.ylabel('Contribution (y)')
plt.title('Michaelis-Menten')
plt.legend()
plt.grid(True)
plt.show()
```
This image uses the typical labels of Michaelis-Menten on Kinetics:
![image](https://github.com/pymc-labs/pymc-marketing/assets/59846724/0b8a256a-8f31-4145-bfb1-5298e5a935f2)